lib/host/wasi/inode-linux.cpp

```diff
@@ -108,6 +108,23 @@ constexpr int openFlags(__wasi_oflags_t OpenFlags, __wasi_fdflags_t FdFlags,
   return Flags;
 }
 
+std::pair<const char *, std::unique_ptr<char[]>>
+createNullTerminatedString(std::string_view View) noexcept {
+  const char *CStr = nullptr;
+  std::unique_ptr<char[]> Buffer;
+  if (!View.empty()) {
+    if (const auto pos = View.find_first_of('\0');
+        pos != std::string_view::npos) {
+      CStr = View.data();
+    } else {
+      Buffer = std::make_unique<char[]>(View.size() + 1);
```

Comment:
Will this make the last char be `\0`?

Reply:
https://stackoverflow.com/questions/42140212/does-make-unique-value-initializes-char-array